bayes_opt/domain_reduction.py

```diff
@@ -96,46 +99,101 @@ def _update(self, target_space: TargetSpace) -> None:
             np.abs(self.current_d) * (self.gamma - self.eta)
 
         self.r = self.contraction_rate * self.r
+        
+    def _windowed_max(self, target_space: TargetSpace) -> np.array:
```

Comment:
I think it would be better to ensure that `target_space.max()` respects the pbounds and then use that function to get the maximum. Otherwise we will have two functions doing pretty much the same thing, which adds maintenance overhead (and the SDRT should really respect constraints, too).

Reply:
This has been implemented. Thanks for the suggestion. I like it better